pallets/gear/src/lib.rs

```diff
@@ -661,13 +659,17 @@ pub mod pallet {
                 Error::<T>::GasLimitTooHigh
             );
 
-            let mut data = Vec::new();
-            // TODO #512
-            code.encode_to(&mut data);
-            salt.encode_to(&mut data);
+            let id: H256 = {
```

Comment:
can we reuse `generate_program_id` func here?

Reply:
Just to unify it's generation everywhere

Reply:
I think it would be nice.

Reply:
done